src/Compilers/CSharp/Portable/Binder/LocalScopeBinder.cs

```diff
@@ -158,7 +158,11 @@ protected ImmutableArray<LocalSymbol> BuildLocals(SyntaxList<StatementSyntax> st
                     case SyntaxKind.DeconstructionDeclarationStatement:
                         {
                             var decl = (DeconstructionDeclarationStatementSyntax)innerStatement;
-                            CollectLocalsFromDeconstruction(decl.Assignment.VariableComponent, LocalDeclarationKind.RegularVariable, locals);
+                            CollectLocalsFromDeconstruction(
+                                decl.Assignment.VariableComponent,
+                                LocalDeclarationKind.RegularVariable,
+                                locals,
+                                innerStatement);
```

Comment:
>innerStatement [](start = 32, length = 14)

Here is where we record the syntax for deconstruction, so we don't have to search for it later.